keps/sig-node/2008-forensic-container-checkpointing/README.md

```diff
@@ -111,6 +118,43 @@ For the first implementation we do not want to support restore in the
 outside of Kubernetes. The restore is a container engine only operation
 in this first step.
 
+A high level view on the implementation is that triggering the *kubelet* API
+endpoint will trigger the `ContainerCheckpoint` CRI API endpoint to create a
+checkpoint at the location defined by the *kubelet*.  In the checkpoint request
+the kubelet will specify the name of the checkpoint archive as
+`checkpoint-<podFullName>-<containerName>-<timestamp>.tar` and also request to
+store the checkpoint archive in the `checkpoints` directory below its root
+directory (as defined by `--root-dir`). This defaults to
+`/var/lib/kubelet/checkpoints`.
+
+To trigger a checkpoint following HTTP Request has to be made against the *kubelet*:
+
+- `POST /checkpoint/{namespace}/{pod}/{container}``
+- Parameters
+  - namespace (in path): string, required, Namespace
+  - pod (in path): string, required, Pod
+  - container (in path): string, required, Container
+  - timeout (in query): integer, Timeout in seconds to wait until the checkpoint
+  creation is finished. If zero or no timeout is specified the default CRI
+  timeout value will be used. Checkpoint creation time depends directly on the
+  used memory of the container. The more memory a container uses the more time
+  is required to create the corresponding checkpoint.
+- Response
+  - 200: OK
+  - 401: Unauthorized
+  - 404: Not Found (if the ContainerCheckpoint feature gate is disabled)
+  - 404: Not Found (if the specified namespace, pod or container cannot be found)
+  - 500: Internal Server Error (if the CRI implementation encounter an error during checkpointing (see error message for further details))
+  - 500: Internal Server Error (if the CRI implementation does not implement the checkpoint CRI API (see error message for further details))
+
+The kubelet APIs are usually restricted to cluster admins and is only accessible
+via `localhost`. Users will not have access to this for now. If, in the future
```

Comment:
Recall that kubelet APIs are reachable via the kube-apiserver API using the node/proxy endpoint, so knowing which permission is used is important.